python/ray/serve/drivers.py

```diff
@@ -83,16 +86,56 @@ async def __call__(self, request: starlette.requests.Request):
 
 @PublicAPI(stability="beta")
 @serve.deployment(route_prefix="/")
-class DAGDriver(SimpleSchemaIngress):
+class DAGDriver:
     def __init__(
         self,
-        dag_handle: RayServeDAGHandle,
-        *,
+        dags: Union[RayServeDAGHandle, Dict[str, RayServeDAGHandle]],
         http_adapter: Optional[Union[str, Callable]] = None,
     ):
-        self.dag_handle = dag_handle
-        super().__init__(http_adapter)
+        install_serve_encoders_to_fastapi()
+        http_adapter = _load_http_adapter(http_adapter)
+        self.app = FastAPI()
+
+        if isinstance(dags, dict):
+            self.dags = dags
+            for route, handle in dags.items():
+
+                def endpoint_create(handle):
+                    @self.app.get(f"{route}")
+                    @self.app.post(f"{route}")
+                    async def handle_request(
+                        request: starlette.requests.Request, inp=Depends(http_adapter)
+                    ):
+                        return await handle.remote(inp)
+
+                endpoint_create_func = functools.partial(endpoint_create, handle)
+                endpoint_create_func()
+
+        else:
+            assert isinstance(dags, (RayServeDAGHandle, RayServeLazySyncHandle))
+            self.dag_handle = dags
+
+            # Single dag case, we will receive all prefix route
+            @self.app.get("/{path:path}")
+            @self.app.post("/{path:path}")
+            async def handle_request(inp=Depends(http_adapter)):
+                return await self.predict(inp)
+
+    async def __call__(self, request: starlette.requests.Request):
+        # NOTE(simon): This is now duplicated from ASGIAppWrapper because we need to
+        # generate FastAPI on the fly, we should find a way to unify the two.
+        sender = ASGIHTTPSender()
+        await self.app(request.scope, receive=request.receive, send=sender)
+        return sender.build_asgi_response()
 
     async def predict(self, *args, **kwargs):
         """Perform inference directly without HTTP."""
         return await self.dag_handle.remote(*args, **kwargs)
+
+    async def multi_dag_predict(self, route_path, *args, **kwargs):
```

Comment:
no longer used?

Reply:
this is used for the handle ingress prediction, we still need a way to select correct handle.

Reply:
ah i see. I do feel this API is very unnatural to use. In that case, we should encourage user to just get a handle for their DAG entrypoint... 
How about we support a special kwargs route_prefix for now? so user we using the same method call just specifying it via `handle.remote(..., route="...")`
Alternatively, a better name might help, how about `predict_with_route`?

Reply:
I will go for the predict_with_route. it is much clear for user to know I will select a handle by sending a route. 

For user behavior, remote(xxx) are all including the inputs (args and kwargs) for their task, add `route` for internal usage is not very clean interface.